File: core/src/main/java/io/nirahtech/erp/core/Human.java
package io.nirahtech.erp.core;

import java.time.LocalDate;

public abstract class Human implements Entity {
    private final String firstName;
    private final String lastName;
    private final LocalDate birthDate;
    private final Gender gender;

    protected Human(final String firstName, final String lastName, final LocalDate birthDate, final Gender gender) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.gender = gender;
    }
    public final String getFirstName() {
        return this.firstName;
    }

    public final String getLastName() {
        return this.lastName;
    }

    public final LocalDate getBirthDate() {
        return this.birthDate;
    }
    public final Gender getGender() {
        return this.gender;
    }
}

File: webapp/src/main/react/src/data/Human.ts
import Gender from "./Gender";

class Human {
    private readonly firstName: string;
    private readonly lastName: string;
    private readonly birthDate: Date;
    private readonly gender: Gender;
  
    constructor(firstName: string, lastName: string, birthDate: Date, gender: Gender) {
      this.firstName = firstName;
      this.lastName = lastName;
      this.birthDate = birthDate;
      this.gender = gender;
    }
  
    getFirstName(): string {
      return this.firstName;
    }
  
    getLastName(): string {
      return this.lastName;
    }
  
    getBirthDate(): Date {
      return this.birthDate;
    }
  
    getGender(): Gender {
      return this.gender;
    }

    is(other: Human): boolean {
      return (other.getFirstName().toLowerCase() == this.firstName.toLowerCase()) && (other.getLastName().toLowerCase() == this.lastName.toLowerCase()) && (other.getBirthDate().toString() == this.birthDate.toString()) && (other.getGender() == this.gender);
    }
  }
export default Human;  